src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -1042,6 +1042,90 @@ TEST_CASE("custom_json_data: handling is_shape_tensor")
       expect_thread_failure);
 }
 
+TEST_CASE("custom_json_data: handling missing optional is_shape_tensor")
+{
+  // Test the case where is_shape_tensor is true and is_optional_ is true
+  // and data for that input is completely omitted
+  PerfAnalyzerParameters params{};
+  params.user_data = {"fake_file.json"};
+  bool is_sequence_model{false};
+
+  std::vector<std::vector<int32_t>> expected_results;
+  std::vector<ModelTensor> tensors;
+  bool expect_init_failure = false;
+  bool expect_thread_failure = false;
+
+  ModelTensor model_tensor1{};
+  model_tensor1.datatype_ = "INT32";
+  model_tensor1.is_optional_ = false;
+  model_tensor1.is_shape_tensor_ = false;
+  model_tensor1.name_ = "INPUT1";
+  model_tensor1.shape_ = {1};
+
+  ModelTensor model_tensor2 = model_tensor1;
+  model_tensor2.name_ = "INPUT2";
+
+  std::string json_str{R"({
+   "data": [
+     { "INPUT2": [21] },
+     { "INPUT2": [22] },
+     { "INPUT2": [23] }     
+   ]})"};
+
+  model_tensor1.is_shape_tensor_ = true;
```

Comment:
The ordering of these assignments is strange. Why not set it to true up there and change tensor 2?

Reply:
done